platform/api.scripting/src/org/netbeans/modules/scripting/GraalEnginesProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.scripting;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.script.ScriptEngineFactory;
+import org.graalvm.polyglot.Engine;
+import org.graalvm.polyglot.Language;
+import org.netbeans.spi.scripting.EngineProvider;
+import org.openide.util.lookup.ServiceProvider;
+
+@ServiceProvider(service = EngineProvider.class)
```

Comment:
Shouldn't the Graal Engine Provider be in a separate module?

Reply:
It used to be.  It got merged in 818830a after in office discussion with Sváťa and Tomáš Zezula.

The needed `org.graalvm.polyglot` types are provided on bootclasspath when running on the GraalVM. Having them in the same module is similar and seemed OK as no additional runtime cost/weight of a module is paid.

Reply:
I find this wrong somehow. So you manage to compile those classes importing `org.graalvm.*` because you have a compile-time dependency on the Graal SDK but then I assume you rely that those classes will never be loaded by a module system classloader unless on the Graal VM?

Having compiled code that cannot be loaded at runtime feels like a broken dependency to me...

Reply:
Right. If we ever want to bundle [Graal.js](https://github.com/graalvm/graaljs) and use it on regular (e.g. non-[GraalVM](http://www.graalvm.org/)) **JVM** then this setup would be wrong. So far I was only testing on either regular **JVM** (the engine missing and disabled) or **GraalVM** (the API classes and the engine present).

Reply:
I still find this an abuse of the module system. Why compile the code is this weird way, add a compile time dependency on Graal and have .class files that are un-loadable when we could have a separate module for people to load when Graal is there?

Maybe have a token or something for GraalVM (something for `OpenIDE-Module-Java-Dependencies` perhaps) ?

Reply:
Hello @emilianbold (and also @jlahoda): per your request I moved the GraalVM specific part (including the documentation) into `libs.graalsdk` module. Now I expect your approvals. Thank you.